servicetests/tests/ShadowUpdate/main.cpp

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/JsonObject.h>
+#include <aws/crt/UUID.h>
+#include <aws/crt/io/HostResolver.h>
+
+#include <aws/crt/mqtt/Mqtt5Packets.h>
+#include <aws/iot/Mqtt5Client.h>
+#include <aws/iot/MqttClient.h>
+
+#include <aws/iotshadow/ErrorResponse.h>
+#include <aws/iotshadow/IotShadowClient.h>
+#include <aws/iotshadow/ShadowDeltaUpdatedEvent.h>
+#include <aws/iotshadow/ShadowDeltaUpdatedSubscriptionRequest.h>
+#include <aws/iotshadow/UpdateNamedShadowRequest.h>
+#include <aws/iotshadow/UpdateShadowRequest.h>
+#include <aws/iotshadow/UpdateShadowResponse.h>
+#include <aws/iotshadow/UpdateShadowSubscriptionRequest.h>
+
+#include <aws/iotshadow/GetShadowRequest.h>
+#include <aws/iotshadow/GetShadowResponse.h>
+#include <aws/iotshadow/GetShadowSubscriptionRequest.h>
+#include <aws/iotshadow/UpdateShadowSubscriptionRequest.h>
+
+#include <algorithm>
+#include <condition_variable>
+#include <iostream>
+#include <mutex>
+#include <thread>
+
+#include "../../../samples/utils/CommandLineUtils.h"
+
+using namespace Aws::Crt;
+using namespace Aws::Iotshadow;
+
+void changeShadowValue(Aws::Crt::String thingName, String property, String value);
+void changeNamedShadowValue(String thingName, String property, String value, String shadowName);
+
+IotShadowClient *shadowClient = nullptr;
+int main(int argc, char *argv[])
+{
+    printf("Starting shadow update test\n");
+    /************************ Setup ****************************/
+    ApiHandle apiHandle;
+
+    /**
+     * cmdData is the arguments/input from the command line placed into a single struct for
+     * use in this sample. This handles all of the command line parsing, validating, etc.
+     * See the Utils/CommandLineUtils for more information.
+     */
+    Utils::cmdData cmdData = Utils::parseSampleInputShadow(argc, argv, &apiHandle);
+
+    /**
+     * In a real world application you probably don't want to enforce synchronous behavior
+     * but this is a sample console application, so we'll just do that with a condition variable.
+     */
+    std::promise<bool> connectionCompletedPromise;
+    std::promise<void> connectionClosedPromise;
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> connection;
+
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithMtlsFromPath(
```

Comment:
This test seemingly supports both 3 and 5 and yet we have a 5 builder unconditionally created and just sitting at top-level scope.

Can we partition the setup out into a create_client helper which in turn conditionally invokes a create_3_based_client and create_5_based_client helper as appropriate?